vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridDetailsRowIT.java

```diff
@@ -146,11 +145,8 @@ private void assertElementHasButton(WebElement componentRenderer,
         Assert.assertEquals(content, children.get(0).getText());
     }
 
-    private void assertElementHasNoButton(WebElement componentRenderer) {
-
-        List<WebElement> children = componentRenderer
-                .findElements(By.tagName("vaadin-button"));
-        Assert.assertEquals(0, children.size());
+    private void assertElementNotDisplayed(WebElement componentRenderer) {
+        Assert.assertFalse(componentRenderer.isDisplayed());
```

Comment:
The tests were relying on (unintentional) row selection to result in the hidden details cell content to get emptied. Fixed the tests to check for the flow-component-renderer visibility instead, and disabled selection in the test page to not have the tests depend on a false premise.